src/Silverback.Integration.MQTT/Messaging/Broker/Mqtt/MqttClientWrapper.cs

```diff
@@ -164,6 +173,9 @@ private async Task<bool> TryConnectAsync(bool isFirstTry, CancellationToken canc
             }
 
             _connectingTaskCompletionSource.SetResult(true);
+
+            ClientConfig.OnConnected?.Invoke(ClientConfig);
```

Comment:
I had to add it here because if it is called before L175 the application gets into an infinite loop when we publish a message in the callback.